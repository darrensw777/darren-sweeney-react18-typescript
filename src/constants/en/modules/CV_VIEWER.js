export const CV_VIEWER = {
    DOWNLOAD: "Click here to download a PDF of my CV",
    CONTACT: "Contact me",
    PAGE: "Page",
    OF: "of",
    NEXT: "Next",
    PREVIOUS: "Previous",
};
