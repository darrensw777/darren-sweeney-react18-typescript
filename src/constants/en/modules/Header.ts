export const HeaderCopy = {
    headerLinks: [
        {
            linkTo: '/',
            label: 'Home',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
        {
            linkTo: '/about',
            label: 'About',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/work',
            label: 'Work',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/music',
            label: 'Music',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/videos',
            label: 'Videos',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/gallery',
            label: 'Gallery',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/contact',
            label: 'Contact',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
    ],
};

export default HeaderCopy;
