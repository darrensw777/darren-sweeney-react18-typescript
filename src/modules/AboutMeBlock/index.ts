/* istanbul ignore file */
export { default } from './AboutMeBlock';
