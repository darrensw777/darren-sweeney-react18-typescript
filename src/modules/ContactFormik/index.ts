/* istanbul ignore file */
export { default } from './ContactFormik';
