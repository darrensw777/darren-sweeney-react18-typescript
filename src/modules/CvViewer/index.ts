/* istanbul ignore file */
export { default } from './CvViewer';
