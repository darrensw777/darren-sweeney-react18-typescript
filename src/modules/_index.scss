@import './Header/Header.scss';
@import './Footer/Footer.scss';
@import './PageHeader/page-header.scss';
@import './AboutMeBlock/about-me-block.scss';
@import './CvViewer/cv-viewer.scss';
@import './SoundCloudPlayer/soundcloud-player.scss';
@import './ImagesGallery/images-gallery.scss';
@import './ContactFormik/contact-formik.scss';
@import './InterestLists/interest-lists.scss';