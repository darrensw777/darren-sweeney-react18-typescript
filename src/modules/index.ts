/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
