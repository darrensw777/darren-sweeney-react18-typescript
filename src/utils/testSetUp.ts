import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
