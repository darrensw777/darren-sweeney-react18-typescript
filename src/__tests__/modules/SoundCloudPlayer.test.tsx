import { SoundCloudPlayer } from 'modules';
import testSetUp from 'utils/testSetUp';

describe('SoundCloudPlayer module', () => {
    const soundcloudIds = [1252449361, 246065520];

    let asFragment: any, container: any;
    beforeEach(() => {
        ({ asFragment, container } = testSetUp({
            Component: <SoundCloudPlayer soundcloudIds={soundcloudIds} />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the player container', () => {
        const player = container.getElementsByClassName('soundcloud-player');
        expect(player.length).toBe(1);
    });

    test('renders the players', () => {
        const player = container.getElementsByClassName('player');
        expect(player.length).toBe(2);
    });
});
