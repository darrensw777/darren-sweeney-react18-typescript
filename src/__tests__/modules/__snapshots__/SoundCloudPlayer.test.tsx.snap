// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundCloudPlayer module matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="moduleStyles__SoundCloudPlayerWrapper-sc-187oehp-9 hdVbCq"
  >
    <div
      class="inner-content soundcloud-player"
    >
      <div>
        <iframe
          class="player"
          frameborder="no"
          height="166"
          scrolling="no"
          src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/1252449361&amp;color=000000&amp;auto_play=false&amp;hide_related=false&amp;show_artwork=true&show_teaser=false"
          title="music"
          width="100%"
        />
      </div>
      <div>
        <iframe
          class="player"
          frameborder="no"
          height="166"
          scrolling="no"
          src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/246065520&amp;color=000000&amp;auto_play=false&amp;hide_related=false&amp;show_artwork=true&show_teaser=false"
          title="music"
          width="100%"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
